### PR TITLE
add NetCDF

### PR DESCRIPTION
**Pull request by scivision**
Like HDF5, [NetCDF](https://www.unidata.ucar.edu/software/netcdf/) is a popular archival high performance data format used in HPC and all sorts of applications.

### CODE REVIEW

**Comment by scivision** [comment]
@jpakkane et al,
Should the NetCDF test be set to skip without failing when NetCDF not present?
I don't want to keep making the Docker image larger, yet I think it's beneficial to have batteries-included support for widely used libraries and features.
For example, we aren't putting Flang, PGI etc. into Docker.

**Comment by jpakkane** [comment]
> Should the NetCDF test be set to skip without failing when NetCDF not present?

Yes.

**Comment by scivision** [comment]
@jpakkane is there a canonical way to skip these tests without CI erroring? It appears that
```meson
error('MESON_SKIP_TEST')
```
is not quite right as this is what's in the meson.build now for NetCDF.

**Comment by jpakkane** [comment]
That should work. I don't really understand why it does not. You should not need the Cygwin and Darwin checks, because `required: false` should take care of that. Explicit disables are only for the case when the platform ships the dependency but is is known to be broken in CI machines (sadly this happens now and again).

In any case please don't use variables with capital letters. This should use the `foo_dep` style that the rest of the code base uses.

**Comment by scivision** [comment]
Oh, the answer to making `error('MESON_SKIP_TEST')` work for packages not in the CI Docker image is that in run_project_tests.py:skippable() there must be an appropriately placed `if ...: return True`

**Comment by scivision** [comment]
#4991 and this are failing Cygwin gccx64ninja with errors including
```
FAIL: test_default_options_set (__main__.RewriterTests)
  -- Project: rewritetest
```

@jpakkane was something recently changed in master for this in the past few hours?  It seems I and the other PRs should not be having this errro

**Comment by mensinda** [comment]
Yep, that was me. I somehow managed to brake the CI once again...

But seriously, how could this have happened? All builds were passing in my original PR. Anyway, #4994 should fix this.

**Comment by scivision** [comment]
@jpakkane this is ready to merge

File: docs/markdown/Dependencies.md [hunk 1]
```diff
@@ -349,6 +349,17 @@ are not in your path, they can be specified by setting the standard
 environment variables `MPICC`, `MPICXX`, `MPIFC`, `MPIF90`, or
 `MPIF77`, during configuration.
 
+## NetCDF
```
Comment (jpakkane) [inline docs/markdown/Dependencies.md]: This should mention that the feature is only available since 0.50.0

**Comment by jpakkane** [comment]
`[skip ci]` is a bit tricky. It should only be used if the _entire_ pull request is a doc-only fix. In this case it did not run Travis checks even though in reality it should have. I'll squash this to rewrite the message but just something to keep in mind in the future.

**Pull request merged**